ProjFS.Mac/PrjFSKext/VirtualizationRoots.cpp

```diff
@@ -314,7 +325,10 @@ static VirtualizationRootHandle FindRootAtVnode_Locked(vnode_t vnode, uint32_t v
         
         if (rootEntry.rootVNode == vnode && rootEntry.rootVNodeVid == vid)
         {
-            assert(fileId.fsid.val[0] == rootEntry.rootFsid.val[0]);
+            assertf(
+                FsidsAreEqual(fileId.fsid, rootEntry.rootFsid) && fileId.inode == rootEntry.rootInode,
+                "FindRootAtVnode_Locked: matching root vnode/vid but not fsid/inode? vnode %p:%u. rootEntry fsid 0x%x:%x, inode 0x%llx, searching for fsid 0x%x:%x, inode 0x%llx",
+                vnode, vid, rootEntry.rootFsid.val[0], rootEntry.rootFsid.val[1], rootEntry.rootInode, fileId.fsid.val[0], fileId.fsid.val[1], fileId.inode);
```

Comment:
Should we `KextLog_Unslide` this `vnode`?  Or is that not required because this is an alert?

Reply:
Good catch! It doesn't matter for security reasons in an assert, but if we want to match the message from the assert up with kernel logs, the slide is helpful. I'll fix that up.

Reply:
This is now fixed.